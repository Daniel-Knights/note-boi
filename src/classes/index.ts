export * from './appError';
export * from './debounceQueue';
export * from './encryptor';
export * from './fetchBuilder';
export * from './keyStore';
export * from './storage';
export * from './unsyncedNotesManager';
