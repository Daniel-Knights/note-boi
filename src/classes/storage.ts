import { DeletedNote } from '../api';
import { ColourTheme, UpdateStrategy } from '../constant';

export const STORAGE_KEYS_STRING = {
  MENU_WIDTH: 'note-menu-width',
  THEME: 'theme',
  UPDATE_SEEN: 'update-seen',
  UPDATE_STRATEGY: 'update-strategy',
  USERNAME: 'username',
} satisfies Record<StorageKeyString, string>;

export const STORAGE_KEYS_JSON = {
  UNSYNCED: 'unsynced-note-uuids',
} satisfies Record<StorageKeyJson, string>;

export const STORAGE_KEYS = { ...STORAGE_KEYS_STRING, ...STORAGE_KEYS_JSON };

/** `localStorage` convenience methods with stronger typing. */
export class Storage {
  static get<T extends keyof StorageValuesString>(key: T): StorageValues[T] | null {
    return localStorage.getItem(STORAGE_KEYS[key]) as StorageValues[T] | null;
  }

  static set<T extends keyof StorageValuesString>(key: T, value: StorageValues[T]): void {
    return localStorage.setItem(STORAGE_KEYS[key], value as string);
  }

  static remove(key: StorageKey) {
    return localStorage.removeItem(STORAGE_KEYS[key]);
  }

  static clear() {
    return localStorage.clear();
  }

  static getJson<T extends keyof StorageValuesJson>(key: T): StorageValues[T] | null {
    const storedItem = localStorage.getItem(STORAGE_KEYS[key]);

    return storedItem ? JSON.parse(storedItem) : null;
  }

  static setJson<T extends keyof StorageValuesJson>(key: T, value: StorageValues[T]) {
    return localStorage.setItem(STORAGE_KEYS[key], JSON.stringify(value));
  }
}

//// Types

type StorageValues = {
  MENU_WIDTH: string;
  THEME: ColourTheme;
  UNSYNCED: {
    new: string;
    edited: string[];
    deleted: DeletedNote[];
  };
  UPDATE_SEEN: string;
  UPDATE_STRATEGY: UpdateStrategy;
  USERNAME: string;
};

export type StorageKey = keyof StorageValues;

type StorageValuesString = {
  [K in StorageKey as StorageValues[K] extends string ? K : never]: StorageValues[K];
};

type StorageValuesJson = {
  [K in StorageKey as StorageValues[K] extends object ? K : never]: StorageValues[K];
};

export type StorageKeyString = keyof StorageValuesString;
export type StorageKeyJson = keyof StorageValuesJson;
