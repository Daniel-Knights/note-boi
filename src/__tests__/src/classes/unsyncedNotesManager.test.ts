import { UnsyncedNotesManager } from '../../../classes';
import { Storage } from '../../../classes/storage';

describe('UnsyncedNotesManager', () => {
  it('Initializes empty when no storage', () => {
    const manager = new UnsyncedNotesManager();

    assert.strictEqual(manager.new, '');
    assert.strictEqual(manager.edited.size, 0);
    assert.strictEqual(manager.deleted.length, 0);
  });

  it('Initializes with contents from storage', () => {
    Storage.setJson('UNSYNCED', {
      new: 'n1',
      edited: ['e1', 'e2'],
      deleted: [{ uuid: 'd1', deleted_at: 0 }],
    });

    const manager = new UnsyncedNotesManager();

    assert.strictEqual(manager.new, 'n1');
    assert.deepEqual(manager.edited, new Set(['e1', 'e2']));
    assert.deepEqual(manager.deleted, [{ uuid: 'd1', deleted_at: 0 }]);
  });

  it('size', () => {
    const manager = new UnsyncedNotesManager();

    manager.new = 'n';
    manager.edited.add('e');
    manager.deleted.push({ uuid: 'd', deleted_at: 0 });

    assert.strictEqual(manager.size, 2);
  });

  describe('clear', () => {
    it('Clears edited and deleted', () => {
      const manager = new UnsyncedNotesManager();
      const setJsonSpy = vi.spyOn(Storage, 'setJson');

      manager.new = 'n';
      manager.edited.add('e');
      manager.deleted.push({ uuid: 'd', deleted_at: 0 });
      manager.clear();
      manager.clear(false);

      assert.strictEqual(manager.new, 'n');
      assert.strictEqual(manager.edited.size, 0);
      assert.strictEqual(manager.deleted.length, 0);
      expect(setJsonSpy).toHaveBeenCalled();
    });

    it('Clears new, edited, and deleted', () => {
      const manager = new UnsyncedNotesManager();
      const removeSpy = vi.spyOn(Storage, 'remove');

      manager.new = 'n';
      manager.edited.add('e');
      manager.deleted.push({ uuid: 'd', deleted_at: 0 });
      manager.clear(true);

      assert.strictEqual(manager.new, '');
      assert.strictEqual(manager.edited.size, 0);
      assert.strictEqual(manager.deleted.length, 0);
      expect(removeSpy).toHaveBeenCalled();
    });
  });

  describe('set', () => {
    it('Updates new, edited, and deleted', () => {
      const manager = new UnsyncedNotesManager();
      const setJsonSpy = vi.spyOn(Storage, 'setJson');

      manager.set({
        new: 'n1',
        edited: ['e1'],
        deleted: [{ uuid: 'd1', deleted_at: 0 }],
      });

      manager.set({
        new: 'n2',
        edited: ['e2'],
        deleted: [{ uuid: 'd2', deleted_at: 0 }],
      });

      assert.strictEqual(manager.new, 'n2');
      assert.deepEqual(manager.edited, new Set(['e1', 'e2']));
      assert.deepEqual(manager.deleted, [
        { uuid: 'd1', deleted_at: 0 },
        { uuid: 'd2', deleted_at: 0 },
      ]);
      expect(setJsonSpy).toHaveBeenCalled();
    });

    it('Removes edited note if also deleted', () => {
      const manager = new UnsyncedNotesManager();

      manager.set({ edited: ['e1', 'e2'] });
      manager.set({ deleted: [{ uuid: 'e1', deleted_at: 0 }] });

      assert.deepEqual(manager.edited, new Set(['e2']));
      assert.deepEqual(manager.deleted, [{ uuid: 'e1', deleted_at: 0 }]);
    });

    it('Resets new if in edited or deleted', () => {
      const manager = new UnsyncedNotesManager();

      manager.set({ new: 'n1' });
      manager.set({ edited: ['n1'] });

      assert.strictEqual(manager.new, '');

      manager.set({ new: 'n2' });
      manager.set({ deleted: [{ uuid: 'n2', deleted_at: 0 }] });

      assert.strictEqual(manager.new, '');
    });
  });

  describe('store', () => {
    it('Removes UNSYNCED if empty', () => {
      const manager = new UnsyncedNotesManager();
      const removeSpy = vi.spyOn(Storage, 'remove');
      const setJsonSpy = vi.spyOn(Storage, 'setJson');

      manager.store();

      expect(removeSpy).toHaveBeenCalledWith('UNSYNCED');
      expect(setJsonSpy).not.toHaveBeenCalled();
    });

    it('Sets UNSYNCED if not empty', () => {
      const manager = new UnsyncedNotesManager();
      const setJsonSpy = vi.spyOn(Storage, 'setJson');

      manager.new = 'n';
      manager.edited.add('e');
      manager.deleted.push({ uuid: 'd', deleted_at: 0 });
      manager.store();

      expect(setJsonSpy).toHaveBeenCalledWith('UNSYNCED', {
        new: 'n',
        edited: ['e'],
        deleted: [{ uuid: 'd', deleted_at: 0 }],
      });
    });
  });
});
