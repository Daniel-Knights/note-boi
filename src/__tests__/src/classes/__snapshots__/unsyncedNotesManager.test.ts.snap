// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnsyncedNotesManager > Initializes empty when no storage > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > Initializes empty when no storage > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > Initializes with contents from storage > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > clear > Clears edited and deleted > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > clear > Clears new, edited, and deleted > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > set > Removes edited note if also deleted > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > set > Resets new if in edited or deleted > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > set > Updates new, edited, and deleted > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > size > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > size > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > size > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > size > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > size > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > size > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > size > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > size > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > size > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > size > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > size > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > size > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > size > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > size > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > store > Removes UNSYNCED if empty > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > KeyStore.getKey() 1`] = `undefined`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > Storage.get('MENU_WIDTH') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > Storage.get('THEME') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > Storage.get('UPDATE_SEEN') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > Storage.get('UPDATE_STRATEGY') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > Storage.get('USERNAME') 1`] = `null`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > allCalls 1`] = `[]`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > document.body.outerHTML 1`] = `"<body class="theme--system"></body>"`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > mockDb 1`] = `
{
  "deletedNotes": undefined,
  "encryptedNotes": [],
  "users": {
    "d": "1",
  },
}
`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > noteState 1`] = `
{
  "extraSelectedNotes": [],
  "notes": [],
  "selectedNote": {
    "content": {
      "body": "",
      "delta": {
        "ops": [],
      },
      "title": "",
    },
    "id": "id",
    "timestamp": 0,
  },
}
`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > openedPopup 1`] = `undefined`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > selectedTheme 1`] = `"System"`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > syncState 1`] = `
{
  "appError": AppError {
    "code": 0,
    "display": undefined,
    "message": undefined,
    "originalError": undefined,
    "retry": [Function],
    "retryConfig": undefined,
  },
  "encryptedNotesCache": Map {},
  "isLoggedIn": false,
  "loadingCount": 0,
  "newPassword": "",
  "password": "",
  "unsyncedNotes": UnsyncedNotesManager {
    "deleted": [],
    "edited": Set {},
    "new": "",
  },
  "username": "",
}
`;

exports[`UnsyncedNotesManager > store > Sets UNSYNCED if not empty > updateState 1`] = `
{
  "isAvailable": false,
  "isDownloading": false,
  "strategy": "manual",
}
`;
